src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Utility/GetPackageReferenceUtility.cs

```diff
@@ -60,38 +60,30 @@ internal static PackageIdentity UpdateResolvedVersion(LibraryDependency projectL
         }
 
         /// <summary>
-        /// Gets the dependencies of a top level package and caches these if the assets file has changed
+        /// Updates the transitive dependencies for the project.
         /// </summary>
-        /// <param name="library">Library from the project file.</param>
-        /// <param name="targetFramework">Target framework from the project file.</param>
-        /// <param name="targets">Target assets file with the package information.</param>
+        /// <param name="libraries">Libraries from the project file.</param>
         /// <param name="installedPackages">Cached installed package information</param>
         /// <param name="transitivePackages">Cached transitive package information</param>
-        internal static IReadOnlyList<PackageIdentity> UpdateTransitiveDependencies(LockFileTargetLibrary library, NuGetFramework targetFramework, IList<LockFileTarget> targets, Dictionary<string, ProjectInstalledPackage> installedPackages, Dictionary<string, ProjectInstalledPackage> transitivePackages)
+        internal static IReadOnlyList<PackageIdentity> UpdateTransitiveDependencies(IList<LockFileTargetLibrary> libraries, Dictionary<string, ProjectInstalledPackage> installedPackages, Dictionary<string, ProjectInstalledPackage> transitivePackages)
```

Comment:
This change is key for the PR, with previous [Andy's comment](https://github.com/NuGet/NuGet.Client/pull/5926#discussion_r1681820594) I was able to remove projects from the libraries so this function return's only transitive packages.

Assets file already contains a flat list of all the dependencies for a project, there was no need to try to create a graph.